cleverpush/src/main/java/com/cleverpush/DeviceIdBroadcastReceiver.java

```diff
@@ -7,21 +7,16 @@
 import android.preference.PreferenceManager;
 
 public class DeviceIdBroadcastReceiver extends BroadcastReceiver {
+    String deviceId;
     @Override
     public void onReceive(Context context, Intent intent) {
-        getActionName(Constants.APPLICATION_GROUP_NAME);
         SharedPreferences sharedPreferences = getSharedPreferences(context);
-        if (intent.getAction().equals(getActionName(Constants.APPLICATION_GROUP_NAME)) && intent.hasExtra(Constants.DEVICE_ID)) {
-            sharedPreferences.edit().putString(CleverPushPreferences.DEVICE_ID, intent.getStringExtra(Constants.DEVICE_ID)).apply();
+        if (intent.getAction().equals(Constants.DEVICE_ID_ACTION_KEY) && intent.hasExtra(Constants.DEVICE_ID)) {
+            deviceId = intent.getStringExtra(Constants.DEVICE_ID);
```

Comment:
define the variable inside of the function

Reply:
Test comment